marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/keystore/AzureKeyStoreInit.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2012-2020 MarkLogic Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package com.marklogic.hub.central.keystore;
+
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobClientBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.io.InputStream;
+
+@Component
+@Profile("azure")
+public class AzureKeyStoreInit extends AbstractKeyStoreInit {
+
+    @Value("${key-store-password}")
```

Comment:
Shouldn't this be e.g. "azure.storage.keyStorePassword"? It's the password that's specific to the keystore for Azure, right?

Reply:
Azure KeyVault stores secrets as a key-value pair.
Here the key is `key-store-password` and the value is the actual password.
Azure does not allow `.` in key names so this is what I've used. 

Also, the key name is kept consistent across both Azure and AWS services.
For AWS we'll fetch value for the key `key-store-password` from the AWS parameter store.

Reply:
Got it, makes sense.